Task07/boxplotChart/boxplotChartPlotly.js

```diff
@@ -0,0 +1,221 @@
+"use strict";
+
+const urlBase = 'http://localhost:3000'
+
+// FUNCTIONS TO GET X COLORS FROM X TO X : GRADIENT
+function hex (c) {
+  let s = "0123456789abcdef";
+  let i = parseInt (c);
+  if (i == 0 || isNaN (c))
+    return "00";
+  i = Math.round (Math.min (Math.max (0, i), 255));
+  return s.charAt ((i - i % 16) / 16) + s.charAt (i % 16);
+}
+/* Convert an RGB triplet to a hex string */
+function convertToHex (rgb) {
+  return hex(rgb[0]) + hex(rgb[1]) + hex(rgb[2]);
+}
+/* Remove '#' in color hex string */
+function trim (s) { return (s.charAt(0) == '#') ? s.substring(1, 7) : s }
+/* Convert a hex string to an RGB triplet */
+function convertToRGB (hex) {
+  let color = [];
+  color[0] = parseInt ((trim(hex)).substring (0, 2), 16);
+  color[1] = parseInt ((trim(hex)).substring (2, 4), 16);
+  color[2] = parseInt ((trim(hex)).substring (4, 6), 16);
+  return color;
+}
+function generateColor(colorStart,colorEnd,colorCount){
+	// The beginning of your gradient
+	let start = convertToRGB (colorStart);
+	// The end of your gradient
+	let end   = convertToRGB (colorEnd);
+	// The number of colors to compute
+	let len = colorCount;
+	//Alpha blending amount
+	let alpha = 0.0;
+	let saida = [];
+	for (let i = 0; i < len; i++) {
+		var c = [];
+		alpha += (1.0/len);
+		c[0] = start[0] * alpha + (1 - alpha) * end[0];
+		c[1] = start[1] * alpha + (1 - alpha) * end[1];
+		c[2] = start[2] * alpha + (1 - alpha) * end[2];
+		saida.push(convertToHex (c));
+	}
+	return saida;
+}
+
+function makeSimpleBoxplotSensor(sensorId=1, nameVar="xxx", nameInst="yyy", nameGateway="") {
+  let timestampNow = Date.now();
+  let timestamp30d = timestampNow - (30*24*60*60*1000) // -30 days
+  let myURL = urlBase + '/Sensors/'+ sensorId +'/datas?start=' + timestamp30d + '&end=' + timestampNow;
+  let myX = [];
+  let myY = [];
+  let myTrace = [];
+  Plotly.d3.json(myURL, function(error, data) {
+    if (data.data.length > 1){
+      let byday={};
+      function groupday(x, index, array)
+      {
+          let d = new Date(x['createdAt']);
+          d = Math.floor(d.getTime()/(1000*60*60*24)); // *24
+          byday[d]=byday[d]||[];
+          byday[d].push({value:x['value'], createdAt:d*(1000*60*60*24)}); // x['createdAt']
+      }
+      data.data.map(groupday); // old PR : data.map(groupday)
+      // console.log(byday);
+      // let txx = new Date(byday['18348'][1]['createdAt']);
+      // console.log( new Date(Math.floor(txx.getTime()/(1000*60*60*24)) * (1000*60*60*24)) );
+
+      let boxColors = generateColor('#ff1100', '#060e7a', 100);
+      // console.log(boxColors);
+
+      const objKeyByDay = Object.keys(byday);
+
+      const minTemp = data.data.reduce((min, p) => p.value < min ? p.value : min, data.data[0].value);
+      const maxTemp = data.data.reduce((max, p) => p.value > max ? p.value : max, data.data[0].value);
+
+      for (const i of objKeyByDay){
```

Comment:
`i`, `j`, etc, doivent être évités sauf lorsqu'ils ont une portée réduite : ces noms de variables portent à confusion car ils ne sont pas parlants.

Préférer un nom complet, ou éventuellement la première lettre du nom de variable de l'objet itéré.

Reply:
ok